wpimath/src/main/java/edu/wpi/first/math/MathUtil.java

```diff
@@ -13,6 +13,39 @@ private MathUtil() {
     throw new AssertionError("utility class");
   }
 
+  /**
+   * Computes the linear interpolation between a and b if t ∈ [0, 1) and the linear extrapolation
+   * otherwise.
+   *
+   * @param a The start value.
+   * @param b The end value.
+   * @param t The fraction for interpolation.
+   * @return The interpolated value.
+   */
+  public static double lerp(double a, double b, double t) {
+    return a + (b - a) * t;
+  }
+
+  /**
+   * Returns the interpolant t that reflects where q is with respect to the range (a, b). In other
+   * words, returns t such that q = a + (b - a)t. If a = b, then returns 0.
+   *
+   * @param a Lower part of interpolation range.
+   * @param b Upper part of interpolation range.
+   * @param q Query.
+   * @return Interpolant.
+   */
+  public static double inverseLerp(double a, double b, double q) {
+    // q = a + (b − a)t
+    // (b − a)t = q − a
+    // t = (q − a)/(b − a)
+    if (a == b) {
```

Comment:
Any concerns about numerical stability when `a - b` is very small but not exactly 0?